docs/developer/watcher-local-test.md

```diff
@@ -0,0 +1,127 @@
+kubectl apply -f https://raw.githubusercontent.com/kyma-project/module-manager/main/config/crd/bases/operator.kyma-project.io_manifests.yaml
+```
+
+### Build and push lifecycle-manager image
+1. Run the following command to get KCP cluster kubeconfig and export it:
+```shell
+k3d kubeconfig get kcp-local > $KUBE_CONFIG_DIR/kcp-local.yaml && \
+export KUBECONFIG=$KUBE_CONFIG_DIR/kcp-local.yaml
+```
+2. Run the following command to build and push lifecycle-manager image to local k3d registry:
+```shell
+export K3D_REG=k3d-registry.localhost:5111 && \
+make -C $KLM_PATH docker-build IMG=$K3D_REG/lifecycle-manager:$KLM_VERSION && \
+make -C $KLM_PATH docker-push IMG=$K3D_REG/lifecycle-manager:$KLM_VERSION
+```
+
+### Deploy lifecycle-manager on KCP:
+Run the following command to deploy lifecycle-manager on the KCP cluster:
+```shell
+export K3D_REG=k3d-registry.localhost:5111 && \
+make -C $KLM_PATH install && \
+make -C $KLM_PATH local-deploy-with-watcher IMG=$K3D_REG/lifecycle-manager:$KLM_VERSION
+```
+
+### Apply sample module templates for sample-kyma:
+Run the following command to deploy lifecycle-manager on the KCP cluster:
+```shell
+export KUBECONFIG=$KUBE_CONFIG_DIR/kcp-local.yaml && \
+kubectl apply -f $KLM_PATH/config/samples/component-integration-installed/operator_v1alpha1_moduletemplate_kcp-module.yaml && \
+kubectl apply -f $KLM_PATH/config/samples/component-integration-installed/operator_v1alpha1_moduletemplate_skr-module.yaml
+```
+### Apply sample Kyma CR
+1. Run the following command to create a local kyma-runtime (SKR) cluster:
+```shell
+k3d cluster create skr-local
+```
+2. Run the following command to generate and apply sample Kyma CR and its corresponding secret on KCP:
+```shell
+export KUBECONFIG=$KUBE_CONFIG_DIR/kcp-local.yaml && \
+cat << EOF | kubectl apply -f -
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kyma-sample
+  namespace: default
+  labels:
+    "operator.kyma-project.io/kyma-name": "kyma-sample"
+    "operator.kyma-project.io/managed-by": "lifecycle-manager"
+data:
+  config: $(echo "${$(k3d kubeconfig get skr-local)/0.0.0.0/host.k3d.internal}" | base64)
+---
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: Kyma
+metadata:
+  name: kyma-sample
+  namespace: default
+spec:
+  channel: regular
+  sync:
+    enabled: true
+  modules:
+    - name: kcp-module
+EOF
+```
+### Verify that watcher-webhook is installed on SKR
+By checking the Kyma Object events, verify that the `SKRWebhookIsReady` ready condition has a `True` Status. 
+```yaml
+Status:                                              
+    Conditions:                                        
+        Message:               skrwebhook is synchronized
+        Observed Generation:   1               
+        Reason:                SKRWebhookIsReady
+        Status:                True
+        Type:                  Ready
+```
+### Edit kyma on SKR
+By adding the following line to the Kyma CR `spec`, trigger the watcher KCP update
```

Comment:
provide a command to easliy switch contexts, e.g. `kubectl config use-context k3d-skr-local` ?